### PR TITLE
[TE] rootcause poc - aggregation endpoint

### PR DESCRIPTION
**Pull request by apucher**
This PR adds the aggregation endpoint for the rootcause POC. The endpoint performs fine-granular dimension aggregation across multiple metrics and time ranges.

### CODE REVIEW

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/v2/AggregationResource.java [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+   * minimal:    curl -X GET 'localhost:1426/aggregation/query?metricIds=1&ranges=1504076400000:1504162800000'
+   * rollup:     curl -X GET 'localhost:1426/aggregation/query?metricIds=1&ranges=1504076400000:1504162800000&rollupCount=5'
+   * multiple:     curl -X GET 'localhost:1426/aggregation/query?metricIds=1,2&ranges=1504076400000:1504162800000,1504176400000:1504262800000&rollupCount=5'
+   * </pre></p>
+   *
+   * @param metricIdsString metric ids separated by ","
+   * @param rangesString time ranges with end exclusive "[start]:[end]" separated by ","
+   * @param filterString (optional) metric filters
+   * @param rollupCount (optional) max number of values per dimensions for rollup
+   * @return Map (keyed by range) of maps (keyed by metric id) of maps (keyed by dimension name) of maps (keyed by dimension value) of values
```
Comment (cyenjung) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/v2/AggregationResource.java]: Keyed by range --> Keyed by time range.

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/v2/AggregationResource.java [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+    return first.append(results);
+  }
+
+  /**
+   * Helper to convert dimension aggregation dataframe to a map of values
+   *
+   * @param df
+   * @return
+   */
+  private static Map<String, Map<String, Double>> dataframeToMap(DataFrame df) {
```
Comment (cyenjung) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/v2/AggregationResource.java]: Add comment: "Map (keyed by dimension name) of maps (keyed by dimension value) of values".

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/v2/AggregationResource.java (hunk 1 above)
Comment (apucher) [reply thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/v2/AggregationResource.java]: fixed

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/v2/AggregationResource.java (hunk 2 above)
Comment (apucher) [reply thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/v2/AggregationResource.java]: fixed

**Pull request merged**